sklearn/datasets/openml.py

```diff
@@ -0,0 +1,200 @@
+import json
+import numbers
+import sys
+import os
+from os.path import join, exists
+from warnings import warn
+
+try:
+    # Python 2
+    from urllib2 import urlopen
+except ImportError:
+    # Python 3+
+    from urllib.request import urlopen
+
+
+from scipy.io.arff import loadarff
+import numpy as np
+
+from .base import get_data_home
+from ..externals.joblib import Memory
+from ..externals.six import StringIO
+from ..externals.six.moves.urllib.error import HTTPError
+from ..utils import Bunch
+
+_SEARCH_NAME = "https://openml.org/api/v1/json/data/list/data_name/{}/limit/1"
+_DATA_INFO = "https://openml.org/api/v1/json/data/{}"
+
+
+def _get_data_info_by_name(name, version):
+    data_found = True
+    try:
+        if version == "active":
+            json_string = urlopen(_SEARCH_NAME.format(name
```

Comment:
Does name need to be escaped??

Reply:
Hm I guess that depends on what symbols are allowed in names...

Reply:
According to the [XSD](https://github.com/openml/OpenML/blob/master/openml_OS/views/pages/api_new/v1/xsd/openml.data.upload.xsd), it should adhere to the following regex: ([a-zA-Z0-9_\-\.\(\),])+